app/src/main/java/com/palone/planahead/data/database/TaskDao.kt

```diff
@@ -0,0 +1,22 @@
+package com.palone.planahead.data.database
+
+import androidx.room.Dao
+import androidx.room.Delete
+import androidx.room.Query
+import androidx.room.Transaction
+import androidx.room.Upsert
+import com.palone.planahead.data.database.task.Task
+import kotlinx.coroutines.flow.Flow
+
+@Dao
+interface TaskDao {
+    @Upsert
+    suspend fun upsert(task: Task): Long
+
+    @Delete
+    suspend fun delete(task: Task)
+
+    @Transaction
+    @Query("SELECT * FROM task ")
```

Comment:
Does this work? Don't you need to do a JOIN here?

Reply:
I was surprised too, but yes, it does work